### PR TITLE
feat(core): append channel to kv key to make it channel aware

### PR DESCRIPTION
**Pull request by jorgemoya**
## What/Why?
We need for kv keys to be channel aware when using multiple locales.

## Testing
Locally each locale/channel has its own kv pairs.

### CODE REVIEW

File: core/lib/kv/keys.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export const STORE_STATUS_KEY = 'storeStatus';
-export const routeCacheKvKey = (pathname: string) => `v3_${pathname}`;
+export const kvKey = (key: string, channelId?: string) =>
+  `${channelId ?? process.env.BIGCOMMERCE_CHANNEL_ID ?? '1'}_${key}`;
```
Comment (deini) [inline core/lib/kv/keys.ts:3]: Thoughts on moving the `KV` namespace logic to this fn? .

File: core/middlewares/with-routes.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ const updateRouteCache = async (pathname: string, event: NextFetchEvent): Promis
     expiryTime: Date.now() + 1000 * 60 * 30, // 30 minutes
   };
 
-  event.waitUntil(kv.set(routeCacheKvKey(pathname), routeCache));
+  event.waitUntil(kv.set(kvKey(`v3_${pathname}`, channelId), routeCache));
```
Comment (deini) [inline core/middlewares/with-routes.ts:79]: Thoughts on moving the key version `v3` to `kvKey()`?

Comment (jorgemoya) [reply core/middlewares/with-routes.ts:79]: Do we need `v3`? 🤔

File: core/lib/kv/keys.ts (hunk 1 above)
Comment (jorgemoya) [reply core/lib/kv/keys.ts:3]: Will do

File: core/middlewares/with-routes.ts (hunk 2 above)
Comment (deini) [reply core/middlewares/with-routes.ts:79]: It's an easy way for us to change all keys if needed. I think we should keep it.

**Review by deini — APPROVED** [review]

**Pull request merged**